lib/membrane_rtc_engine/engine.ex

```diff
@@ -2,17 +2,111 @@ defmodule Membrane.RTC.Engine do
   @moduledoc """
   RTC Engine implementation.
 
-  One RTC Engine instance is responsible for managing one room in which
-  all tracks of one peer are forwarded to all other peers.
+  RTC Engine is an abstraction layer responsible for linking together different types of `Endpoints`.
+  From implementation point of view RTC Engine is a `Membrane.Pipeline`.
+
+  ## Endpoints
+
+  `Endpoints` are `Membrane.Bin`s able to publish own tracks and subscribe for tracks from other Endpoints.
+  One can thought about Endpoint as an entity responsible for handling some specifc task.
+  Endpoint can be added and removed using `add_endpoint/3` and `remove_endpoint/1` respectively.
+
+  There are basically two types of Endpoints:
+  * Standalone Endpoints - they are in most cases spawned only once per RTC Engine instance and they are not associated with any peer.
+  * Peer Endpoints - they are associated with some peer.
+  Associating Endpoint with Peer will cause RTC Engine sending some Media Events to client library
+  e.g. one which indicates which tracks belong to which peer.
+
+  Currently RTC Engine ships with implementation of two Endpoints:
+  * `#{inspect(__MODULE__)}.Endpoint.WebRTC` which is responsible for establishing connection with some WebRTC
+  peer (mainly browser) and exchanging media with it. WebRTC Endpoint is a Peer Endpoint.
+  * `#{inspect(__MODULE__)}.Endpoint.HLS` which is responsible for receiving media tracks from all other Endpoints and
+  saving them to a file by creating HLS playlist. HLS Endpoint is a Standalone Endpoint.
```

Comment:
```suggestion
  saving them to a file by creating HLS playlists. HLS Endpoint is a Standalone Endpoint.
```

Reply:
It's possible to associate HLS Endpoint with some Peer but HLS Endpoint is a typical server side thing. It doesn't make sense to publish it as a peer especially that it doesn't emit any tracks to other Endpoints